### PR TITLE
Generate functions that required in the package.

### PR DESCRIPTION
**Pull request by Tefera19**
Hi @randrescastaneda ,

This PR contains the update I made to the package.

## Summary by CodeRabbit

* **New Features**
  * Added fk_pip_ref(): create synthetic PIP datasets from a reference folder; supports pct sampling and optional parallel processing.

* **Documentation**
  * README and man pages updated with fk_pip_ref() usage, examples (sequential and parallel), and legacy fk_pip() guidance.

* **Tests**
  * New test suite and fixtures validating synthesis, sampling, copying behavior, and edge cases.

* **Dependencies**
  * Suggests updated to include future.apply and testthat (>= 3.0.0).

* **Version**
  * Bumped to 0.3.0.

* **License**
  * License file reference updated.

### CODE REVIEW

File: R/fk_pip_ref.R [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        is_group_bin <- grepl("_(GROUP|BIN)\\.[^.]+$", fname,
+                              ignore.case = FALSE)
+
+        if (is_group_bin) {
+          # Copy directly
+          fs::file_copy(f, fs::path(svy_out, fname), overwrite = TRUE)
+        } else {
+          # Synthesise
+          synth_svy_file(file_path = f, output_dir = svy_out, pct = pct)
+        }
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/fk_pip_ref.R:171]: comment the original code and make changes based on the code suggestion

File: R/synth_helpers.R [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      uvals <- unique(dt[[col]])
+      if (length(uvals) == 1L) {
+        data.table::set(synth_core, j = col, value = rep(uvals, n_obs))
+      } else {
+        # Resample from original, preserving type
+        sampled <- sample(dt[[col]], size = n_obs, replace = TRUE)
+        data.table::set(synth_core, j = col, value = sampled)
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/synth_helpers.R:132]: Keep code suggestion

File: R/synth_helpers.R [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    for (col in other_cols) {
+      uvals <- unique(dt[[col]])
+      if (length(uvals) == 1L) {
+        data.table::set(synth_core, j = col, value = rep(uvals, n_obs))
+      } else {
+        sampled <- sample(dt[[col]], size = n_obs, replace = TRUE)
+        data.table::set(synth_core, j = col, value = sampled)
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/synth_helpers.R:217]: implement the code suggestion
```suggestion
  if (length(other_cols) > 0) {
    for (col in other_cols) {
      uvals <- unique(dt[[col]])
      if (length(uvals) == 1L) {
        data.table::set(synth_core, j = col, value = rep(uvals, n_obs))
      } else {
        sampled <- sample(dt[[col]], size = n_obs, replace = TRUE)
        data.table::set(synth_core, j = col, value = sampled)
      }
    }
  }
```

File: R/fk_pip_ref.R [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    lineup_files <- fs::dir_ls(lineup_dir, type = "file")
+
+    if (length(lineup_files) > 0) {
+
+      cli::cli_progress_bar("Synthesising lineup_data",
+                            total = length(lineup_files))
+
+      for (f in lineup_files) {
+        synth_lineup_file(file_path = f, output_dir = lineup_out, pct = pct)
+        cli::cli_progress_update()
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/fk_pip_ref.R]: the expected file extension in lineup folder is fst, we don't expect files with different extension. So, we will keep the current setting as is.

File: Develop an AI-driven package that proces.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+Step 5: Implement error handling to manage any issues that arise during the processing, such as missing files, mismatched mean values, or issues with the FakeDataR package. The AI should log any errors encountered and provide informative messages to help users understand what went wrong and how to fix it.
+Step 6: Provide documentation and usage instructions for the package, including how to install it, how to configure the output location, and how to run the processing workflow. Include examples of how to use the package with sample data to demonstrate its functionality.
+Step 7: Refractor the code to ensure it is efficient, maintainable, and follows best practices for R programming. This includes organizing the code into functions, using appropriate data structures, and ensuring that the code is well-commented for clarity.
+Step 8: Test the package with various datasets to ensure that it works correctly and produces the expected results.
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply Develop an AI-driven package that proces.md]: this a draft file, we don't need it.

File: README.Rmd [hunk 6]
```diff
@@ -45,16 +45,35 @@ mean_welf
...
 ## For `pipapi`
 
-This package can be used to create fake data to test `pipapi`. You can provide an `input_path` to a reference folder you want to mimic. If `input_path` is not provided the `fk_pip` functions uses the fake data already installed within this package. The following example will save the new fake data into your working directory (`output_path`).
+This package can be used to create fake data to test `pipapi`. There are two approaches:
 
-```{r pipapi_ex}
+### Using `fk_pip_ref()` (recommended)
 
-## basic example code for pipapi using input_path
+`fk_pip_ref()` takes a reference PIP folder and creates a synthetic copy that mirrors its structure. It synthesises `survey_data` and `lineup_data` files using quantile-remapping (preserving weighted means per area/reporting level), copies `_GROUP`/`_BIN` files directly, and copies `_aux`, `estimations`, and root files verbatim. The `pct` argument controls the fraction of rows to generate (e.g. `0.1` for 10%).
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply README.Rmd]: Correct all the files in root folder are not copied. The suggestion should be implemented.

File: README.md [hunk 7]
```diff
@@ -38,19 +38,69 @@ mean_welf
...
+
+### Using `fk_pip_ref()` (recommended)
+
+`fk_pip_ref()` takes a reference PIP folder and creates a synthetic copy
+that mirrors its structure. It synthesises `survey_data` and
+`lineup_data` files using quantile-remapping (preserving weighted means
+per area/reporting level), copies `_GROUP`/`_BIN` files directly, and
+copies `_aux`, `estimations`, and root files verbatim. The `pct`
+argument controls the fraction of rows to generate (e.g. `0.1` for 10%).
+
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply README.md]: suggestion has been implemented.

File: R/fk_pip_ref.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/fk_pip_ref.R:171]: we expect fst files in the survey_data directory. Keep the original code.

File: R/fk_pip_ref.R [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+  }
+
+  # Check if output is a descendant of input (child/grandchild, etc.)
+  if (grepl(paste0("^", utils::glob2rx(paste0(abs_input, "/*"))), abs_output)) {
+    cli::cli_abort("Cannot create synthetic folder: {.arg output_path} cannot be a subdirectory of {.arg input_path}.")
+  }
+
+  # ---- Create output root ------------------------------------------------
+
+  fs::dir_create(output_path)
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/fk_pip_ref.R]: If the output folder exists, update it.

File: R/fk_pip_ref.R [hunk 9]
```diff
@@ -0,0 +1,220 @@
+#' Generate synthetic PIP data from a reference folder
+#'
+#' Takes a reference PIP API folder and creates a synthetic copy that mirrors
+#' its structure. Folders `_aux` and `estimations` are copied verbatim. Root
+#' files are copied except `cache.duckdb`. Survey micro files are synthesised
+#' using quantile-remapping (preserving per-area weighted mean); files ending in
+#' `_GROUP` or `_BIN` are copied directly. Lineup files are synthesised with
+#' distribution-preserving quantile interpolation.
```
[the remark this answers is not in the document]
Comment (Tefera19) [reply R/fk_pip_ref.R]: `_vintage' subfolder should not be copied.
